server/graphql/v2/query/ExpenseQuery.ts

```diff
@@ -22,26 +23,22 @@ const ExpenseQuery = {
     },
   },
   async resolve(_, args, req) {
+    let expense;
     if (args.expense) {
-      const expense = await fetchExpenseWithReference(args.expense, req);
-
-      if (expense?.status === expenseStatus.DRAFT) {
-        const canViewDraftExpense =
-          expense.data?.draftKey === args.draftKey ||
-          req.remoteUser?.isAdmin(expense.FromCollectiveId) ||
-          req.remoteUser?.isAdminOfCollectiveOrHost(await expense.getCollective());
-
-        if (!canViewDraftExpense) {
-          return null;
-        }
-      }
-
-      return expense;
+      expense = await fetchExpenseWithReference(args.expense, req);
     } else if (args.id) {
-      return req.loaders.Expense.byId.load(args.id);
+      expense = await req.loaders.Expense.byId.load(args.id);
     } else {
       throw new Error('You must either provide an id or an expense');
     }
+
+    if (!expense) {
+      return null;
+    } else if (expense.status === expenseStatus.DRAFT && args.draftKey && expense.data?.draftKey === args.draftKey) {
```

Comment:
Maybe this logic can be simplified with:
```suggestion
    if (expense?.status === expenseStatus.DRAFT && args.draftKey && expense?.data?.draftKey === args.draftKey) {
```

Reply:
I tend to prefer the first form, it's more verbose but also more explicit IMO